// ng2-bootstrap/dropdown/bs-dropdown-container.component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */

import * as import0 from 'ng2-bootstrap/dropdown/bs-dropdown-container.component';
import * as import1 from '@angular/core/src/linker/view';
import * as import2 from '@angular/core/src/render/api';
import * as import3 from '@angular/core/src/linker/view_utils';
import * as import4 from '@angular/core/src/metadata/view';
import * as import5 from '@angular/core/src/linker/view_type';
import * as import6 from '@angular/core/src/change_detection/constants';
import * as import7 from '@angular/core/src/linker/component_factory';
import * as import8 from 'ng2-bootstrap/dropdown/bs-dropdown.state';
import * as import9 from '@angular/core/src/change_detection/change_detection_util';
export class Wrapper_BsDropdownContainerComponent {
  /*private*/ _eventHandler:Function;
  context:import0.BsDropdownContainerComponent;
  /*private*/ _changed:boolean;
  constructor(p0:any) {
    this._changed = false;
    this.context = new import0.BsDropdownContainerComponent(p0);
  }
  ngOnDetach(view:import1.AppView<any>,componentView:import1.AppView<any>,el:any):void {
  }
  ngOnDestroy():void {
    this.context.ngOnDestroy();
  }
  ngDoCheck(view:import1.AppView<any>,el:any,throwOnChange:boolean):boolean {
    var changed:any = this._changed;
    this._changed = false;
    return changed;
  }
  checkHost(view:import1.AppView<any>,componentView:import1.AppView<any>,el:any,throwOnChange:boolean):void {
  }
  handleEvent(eventName:string,$event:any):boolean {
    var result:boolean = true;
    return result;
  }
  subscribe(view:import1.AppView<any>,_eventHandler:any):void {
    this._eventHandler = _eventHandler;
  }
}
var renderType_BsDropdownContainerComponent_Host:import2.RenderComponentType = import3.createRenderComponentType('',0,import4.ViewEncapsulation.None,([] as any[]),{});
class View_BsDropdownContainerComponent_Host0 extends import1.AppView<any> {
  _el_0:any;
  compView_0:import1.AppView<import0.BsDropdownContainerComponent>;
  _BsDropdownContainerComponent_0_3:Wrapper_BsDropdownContainerComponent;
  constructor(viewUtils:import3.ViewUtils,parentView:import1.AppView<any>,parentIndex:number,parentElement:any) {
    super(View_BsDropdownContainerComponent_Host0,renderType_BsDropdownContainerComponent_Host,import5.ViewType.HOST,viewUtils,parentView,parentIndex,parentElement,import6.ChangeDetectorStatus.CheckAlways);
  }
  createInternal(rootSelector:string):import7.ComponentRef<any> {
    this._el_0 = import3.selectOrCreateRenderHostElement(this.renderer,'bs-dropdown-container',new import3.InlineArray2(2,'style','display:block;position: absolute;'),rootSelector,(null as any));
    this.compView_0 = new View_BsDropdownContainerComponent0(this.viewUtils,this,0,this._el_0);
    this._BsDropdownContainerComponent_0_3 = new Wrapper_BsDropdownContainerComponent(this.injectorGet(import8.BsDropdownState,this.parentIndex));
    this.compView_0.create(this._BsDropdownContainerComponent_0_3.context);
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [this._el_0]),(null as any));
    return new import7.ComponentRef_<any>(0,this,this._el_0,this._BsDropdownContainerComponent_0_3.context);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import0.BsDropdownContainerComponent) && (0 === requestNodeIndex))) { return this._BsDropdownContainerComponent_0_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    if (this._BsDropdownContainerComponent_0_3.ngDoCheck(this,this._el_0,throwOnChange)) { this.compView_0.markAsCheckOnce(); }
    this.compView_0.internalDetectChanges(throwOnChange);
  }
  destroyInternal():void {
    this.compView_0.destroy();
    this._BsDropdownContainerComponent_0_3.ngOnDestroy();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  visitProjectableNodesInternal(nodeIndex:number,ngContentIndex:number,cb:any,ctx:any):void {
    if (((nodeIndex == 0) && (ngContentIndex == 0))) {  }
  }
}
export const BsDropdownContainerComponentNgFactory:import7.ComponentFactory<import0.BsDropdownContainerComponent> = new import7.ComponentFactory<import0.BsDropdownContainerComponent>('bs-dropdown-container',View_BsDropdownContainerComponent_Host0,import0.BsDropdownContainerComponent);
const styles_BsDropdownContainerComponent:any[] = ([] as any[]);
var renderType_BsDropdownContainerComponent:import2.RenderComponentType = import3.createRenderComponentType('',1,import4.ViewEncapsulation.None,styles_BsDropdownContainerComponent,{});
export class View_BsDropdownContainerComponent0 extends import1.AppView<import0.BsDropdownContainerComponent> {
  _text_0:any;
  _el_1:any;
  _text_2:any;
  /*private*/ _expr_3:any;
  /*private*/ _expr_4:any;
  /*private*/ _expr_5:any;
  /*private*/ _expr_6:any;
  constructor(viewUtils:import3.ViewUtils,parentView:import1.AppView<any>,parentIndex:number,parentElement:any) {
    super(View_BsDropdownContainerComponent0,renderType_BsDropdownContainerComponent,import5.ViewType.COMPONENT,viewUtils,parentView,parentIndex,parentElement,import6.ChangeDetectorStatus.CheckOnce);
    this._expr_3 = import9.UNINITIALIZED;
    this._expr_4 = import9.UNINITIALIZED;
    this._expr_5 = import9.UNINITIALIZED;
    this._expr_6 = import9.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import7.ComponentRef<any> {
    const parentRenderNode:any = this.renderer.createViewRoot(this.parentElement);
    this._text_0 = this.renderer.createText(parentRenderNode,'\n    ',(null as any));
    this._el_1 = import3.createRenderElement(this.renderer,parentRenderNode,'div',import3.EMPTY_INLINE_ARRAY,(null as any));
    this.projectNodes(this._el_1,0);
    this._text_2 = this.renderer.createText(parentRenderNode,'\n  ',(null as any));
    this.init((null as any),((<any>this.renderer).directRenderer? (null as any): [
      this._text_0,
      this._el_1,
      this._text_2
    ]
    ),(null as any));
    return (null as any);
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_3:any = (this.context.direction === 'up');
    if (import3.checkBinding(throwOnChange,this._expr_3,currVal_3)) {
      this.renderer.setElementClass(this._el_1,'dropup',currVal_3);
      this._expr_3 = currVal_3;
    }
    const currVal_4:any = (this.context.direction === 'down');
    if (import3.checkBinding(throwOnChange,this._expr_4,currVal_4)) {
      this.renderer.setElementClass(this._el_1,'dropdown',currVal_4);
      this._expr_4 = currVal_4;
    }
    const currVal_5:any = this.context.isOpen;
    if (import3.checkBinding(throwOnChange,this._expr_5,currVal_5)) {
      this.renderer.setElementClass(this._el_1,'show',currVal_5);
      this._expr_5 = currVal_5;
    }
    const currVal_6:any = this.context.isOpen;
    if (import3.checkBinding(throwOnChange,this._expr_6,currVal_6)) {
      this.renderer.setElementClass(this._el_1,'open',currVal_6);
      this._expr_6 = currVal_6;
    }
  }
}
